roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -1253,7 +1250,7 @@ int RooAbsPdf::calcAsymptoticCorrectedCovariance(RooMinimizer &minimizer, RooAbs
       double prob = getVal(&obs);
       for (int k = 0; k < floated.getSize(); k++) {
          for (int l = 0; l < floated.getSize(); l++) {
-            num(k, l) += data.weight() * data.weight() * diffs[k] * diffs[l] / (prob * prob);
+            num(k, l) += data.weightSquared() * diffs[k] * diffs[l] / (prob * prob);
```

Comment:
For RooDataSet , is `data.weightSquared()` equal to 
`data.weight()^2` ?

Reply:
Yes, right now it is hardcoded to that

Reply:
So this PR only makes a difference for RooDataHist, making it consistent with `SumW2Error` where `weightSquared()` is used as well:
https://github.com/root-project/root/blob/master/roofit/roofitcore/src/RooNLLVar.cxx#L485

Reply:
The comment is outdated now, because the relevant change entered a different PR: https://github.com/root-project/root/pull/11282.